### PR TITLE
awscli2: 2.2.40 -> 2.3.4

### PR DESCRIPTION
**Pull request by tuxinaut**
###### Motivation for this change

Update to the latest version. Please read the **This pr needs some help** section.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review 143689"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

###### This pr needs some help 👩‍⚕️

The tests do not run through :shrug: 

```
ERROR functional/botocore/test_waiter_config.py
ERROR functional/botocore/leak/test_resource_leaks.py::TestDoesNotLeakMemory::test_create_memory_clients_in_loop
ERROR functional/botocore/leak/test_resource_leaks.py::TestDoesNotLeakMemory::test_create_memory_paginators_in_loop
ERROR functional/botocore/leak/test_resource_leaks.py::TestDoesNotLeakMemory::test_create_memory_waiters_in_loop
ERROR functional/botocore/leak/test_resource_leaks.py::TestDoesNotLeakMemory::test_create_single_client_memory_constant
ERROR functional/botocore/leak/test_resource_leaks.py::TestDoesNotLeakMemory::test_create_single_paginator_memory_constant
ERROR functional/botocore/leak/test_resource_leaks.py::TestDoesNotLeakMemory::test_create_single_waiter_memory_constant
= 6 failed, 40674 passed, 4 skipped, 1126 warnings, 8 errors in 503.42s (0:08:23) =
Traceback (most recent call last):
  File "/build/source/scripts/ci/run-tests", line 91, in <module>
```

Therefore, I disabled the tests for now but this should not be necessary.

### CODE REVIEW

**Comment by tuxinaut** [comment]
I'm still searching for help :smile_cat:

File: pkgs/tools/admin/awscli2/default.nix [hunk 1]
```diff
@@ -40,24 +32,17 @@ let
...
   src = fetchFromGitHub {
     owner = "aws";
     repo = "aws-cli";
     rev = version;
-    sha256 = "sha256-IHnNRER9ePKVI9ez15HgxLDR1n6QR0iRESgNqbxQPx8=";
+    sha256 = "sha256-dg9Cv7PQgxqeutwGgo9QNEXoCKv9tZ9o+krJwpivJ/g=";
   };
 
-  patches = [
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/tools/admin/awscli2/default.nix:53]: Please remove the unused fetchpatch from inputs.

File: pkgs/tools/admin/awscli2/default.nix [hunk 2]
```diff
@@ -67,7 +52,7 @@ with py.pkgs; buildPythonApplication rec {
       --replace "distro>=1.5.0,<1.6.0" "distro"
   '';
 
-  checkInputs = [ jsonschema mock nose ];
+  checkInputs = [ jsonschema mock pytest pytest-xdist];
```
Comment (SuperSandro2000) [inline pkgs/tools/admin/awscli2/default.nix:55]: Please use pytestCheckHook.

File: pkgs/tools/admin/awscli2/default.nix [hunk 3]
```diff
@@ -94,7 +79,7 @@ with py.pkgs; buildPythonApplication rec {
     # https://github.com/NixOS/nixpkgs/issues/16144#issuecomment-225422439
     export HOME=$TMP
 
-    AWS_TEST_COMMAND=$out/bin/aws python scripts/ci/run-tests
+    # AWS_TEST_COMMAND=$out/bin/aws python scripts/ci/run-tests
```
Comment (SuperSandro2000) [inline pkgs/tools/admin/awscli2/default.nix:82]:
```suggestion
```

File: pkgs/tools/admin/awscli2/default.nix (hunk 1 above)
Comment (tuxinaut) [reply pkgs/tools/admin/awscli2/default.nix:53]: :heavy_check_mark:

File: pkgs/tools/admin/awscli2/default.nix (hunk 2 above)
Comment (tuxinaut) [reply pkgs/tools/admin/awscli2/default.nix]: :heavy_check_mark:

File: pkgs/tools/admin/awscli2/default.nix (hunk 3 above)
Comment (tuxinaut) [reply pkgs/tools/admin/awscli2/default.nix]: So as you suggested I removed the whole line and thus the tests!

Comment (davegallant) [reply pkgs/tools/admin/awscli2/default.nix]: These tests have been causing periodic issues on darwin also. https://github.com/NixOS/nixpkgs/issues/136214

**Review by davegallant — APPROVED** [review]

**Comment by austinbutler** [comment]
Result of `nixpkgs-review pr 143689` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>awscli2</li>
  </ul>
</details>

**Comment by austinbutler** [comment]
I get:

```
error: builder for '/nix/store/l3j4higb31r703cmg98f90s7c6vprnrc-python3.9-botocore-2.0.0dev155.drv' failed with exit code 1;
       last 10 log lines:
       > adding 'botocore-2.0.0.dev155.dist-info/top_level.txt'
       > adding 'botocore-2.0.0.dev155.dist-info/RECORD'
       > removing build/bdist.macosx-10.6-x86_64/wheel
       > Finished executing setuptoolsBuildPhase
       > installing
       > Executing pipInstallPhase
       > /private/tmp/nix-build-python3.9-botocore-2.0.0dev155.drv-0/source/dist /private/tmp/nix-build-python3.9-botocore-2.0.0dev155.drv-0/source
       > Processing ./botocore-2.0.0.dev155-py3-none-any.whl
       > ERROR: Could not find a version that satisfies the requirement awscrt==0.12.4 (from botocore) (from versions: none)
       > ERROR: No matching distribution found for awscrt==0.12.4
       For full logs, run 'nix log /nix/store/l3j4higb31r703cmg98f90s7c6vprnrc-python3.9-botocore-2.0.0dev155.drv'.
error: 1 dependencies of derivation '/nix/store/bnrn5fjcgx7pb887z52sljgfnxczlkgp-python3.9-s3transfer-0.4.2.drv' failed to build
error: 2 dependencies of derivation '/nix/store/935g6ybxg2xx49q39177fkxxj1h2i4mb-awscli2-2.3.3.drv' failed to build
error: 1 dependencies of derivation '/nix/store/ma7nh29vlj9vikg3i5pn7biabxdf85wx-review-shell.drv' failed to build
```

File: pkgs/tools/admin/awscli2/default.nix [hunk 4]
```diff
@@ -1,22 +1,14 @@
-{ lib, python3, groff, less, fetchFromGitHub, fetchpatch }:
+{ lib, python3, groff, less, fetchFromGitHub }:
 let
   py = python3.override {
     packageOverrides = self: super: {
-      awscrt = super.awscrt.overridePythonAttrs (oldAttrs: rec {
```
Comment (austinbutler) [inline pkgs/tools/admin/awscli2/default.nix:5]: `awscrt` is still required and pinned to an older version than is in `nixpkgs`:
https://github.com/aws/aws-cli/blob/29ee73bc7bb1b0538cb315313cf149b5215ee2dc/setup.cfg#L38
https://github.com/NixOS/nixpkgs/blob/3adcef25692be8ed9575d01766c3ab7b1a26f2cb/pkgs/development/python-modules/awscrt/default.nix#L5

So we either need to leave this override in place, patch `awscli2` to be more relaxed about the version of `awscrt`, or ensure updates to both `awscrt` and `awscli2` are in sync (which currently would mean downgrading the package on master).

**Review by austinbutler — CHANGES_REQUESTED** [review]

File: pkgs/tools/admin/awscli2/default.nix (hunk 4 above)
Comment (austinbutler) [reply pkgs/tools/admin/awscli2/default.nix:5]: With this it builds for me:

```nix
      awscrt = super.awscrt.overridePythonAttrs (oldAttrs: rec {
        version = "0.12.4";
        src = self.fetchPypi {
          inherit (oldAttrs) pname;
          inherit version;
          sha256 = "sha256:1cmfkcv2zzirxsb989vx1hvna9nv24pghcvypl0zaxsjphv97mka";
        };
      });
```

Comment (davegallant) [reply pkgs/tools/admin/awscli2/default.nix:5]: Might make sense to keep it pinned. Looks like [botocore also requires awscrt==0.12.4](https://github.com/boto/botocore/blob/3ff30ef929da20bc83a1f21d1bd3ab6e408648fe/setup.cfg#L9). 

It's 0.12.5 on the [the development branch of botcore](https://github.com/boto/botocore/blob/f272c5b277bc049b221bca40d10ca967d7161dd8/setup.cfg#L11).

File: pkgs/tools/admin/awscli2/default.nix (hunk 3 above)
Comment (tuxinaut) [reply pkgs/tools/admin/awscli2/default.nix]: :+1:

File: pkgs/tools/admin/awscli2/default.nix (hunk 4 above)
Comment (tuxinaut) [reply pkgs/tools/admin/awscli2/default.nix:5]: Thx for the input :+1:

**Pull request merged**